tools/c7n_azure/c7n_azure/resources/sqlserver.py

```diff
@@ -156,56 +157,107 @@ class VulnerabilityAssessmentFilter(Filter):
               - type: vulnerability-assessment
                 enabled: false
 
+    :example:
+
+    Find SQL Servers where vulnerability assessments are not being sent to a
+    required email
+
+    .. code-block:: yaml
+
+        policies:
+          - name: sql-server-no-email
+            resource: azure.sql-server
+            filters:
+              - type: vulnerability-assessment
+                key: recurringScans.emails[?@ == `required@ops.domain`]
+                value: empty
+
+    When using the above value filter form, the data takes the following shape:
+
+    .. code-block:: json
+
+        "storageContainerPath": "https://testznubm7c1.blob.core.windows.net/testznubm7c1/",
+        "recurringScans": {
+            "isEnabled": true,
+            "emailSubscriptionAdmins": false,
+            "emails": [
+                "ops@fake.email",
+                "admins@fake.email"
+            ]
+        }
+
     """
 
     schema = type_schema(
         'vulnerability-assessment',
-        required=['type', 'enabled'],
-        **{
-            'enabled': {"type": "boolean"},
-        }
+        rinherit=ValueFilter.schema,
+        enabled=dict(type='boolean')
     )
 
     log = logging.getLogger('custodian.azure.sqldatabase.vulnerability-assessment-filter')
 
+    def validate(self):
+        # only allow legacy behavior or new ValueFilter behavior, not both
+        # when in "legacy" mode the only entries should be "type" (required by schema) and
+        # "enabled" (required by is_legacy)
+        if self.is_legacy:
+            if len(self.data) > 2:
+                raise PolicyValidationError(
+                    "When using 'enabled', ValueFilter properties are not allowed")
+        # only validate value filter when not in "legacy" mode
+        else:
+            super(VulnerabilityAssessmentFilter, self).validate()
+
     def __init__(self, data, manager=None):
         super(VulnerabilityAssessmentFilter, self).__init__(data, manager)
-        self.enabled = self.data['enabled']
+
+        self.enabled = self.data.get('enabled')
+        # track if we are using the legacy behavior
+        self.is_legacy = 'enabled' in self.data
+        # location on the resource object to store the VA properties
+        self.key = 'c7n:vulnerability_assessment'
```

Comment:
nit: I think we can probably define something like `annotation_key = 'c7n:vulnerability_assessment'` at the class level to be more consistent with other filters and make the attribute name a bit more explicit. This does work as is though, so... nit :)